core/src/main/resources/schema/postgres/postgres-event-tag-migration.sql

```diff
@@ -0,0 +1,30 @@
+-- >>>>>>>>>>> before rolling updates.
+
+ALTER TABLE EVENT_TAG
+    ADD PERSISTENCE_ID  VARCHAR(255),
+    ADD SEQUENCE_NUMBER BIGINT;
+
+-- >>>>>>>>>>> after projection catch up.
```

Comment:
It's not clear to me what do you mean by projection catch-up? Are you referring to the migration tool? The tool is intended for migration from previous versions. 

With the changes in this PR, we need to run another kind of migration.

Luckily, we can do this migration with plain SQL. For example, for Postgres, we can run the following:

```sql
UPDATE event_tag
  SET
    persistence_id = event_journal.persistence_id,
    sequence_number = event_journal.sequence_number
  FROM event_journal
  WHERE event_journal.ordering = event_tag.event_id;
```

After that, we can create the new foreign key and even drop the event_id.

Reply:
"after projection catch-up" refers to the process of waiting for Akka Projection to update the offset to the earliest new column write.

I hadn't considered the option of migrating data using SQL, which is great but I overlooked it. Including this SQL in the migration steps allows us to proceed without waiting for the Projection read "catch up".

I will update PR after fix integration test.

Reply:
For clarification, there is no projection involved here. 

The tags are written when the events are persisted, not when consumed by a Projection. 

So basically, the journal stays what it is. Old tags will be written with ordering filled and pid and seq_nr will stay empty. New tags will have pid and seq_nr filled. Without explicit migration, nothing will happen.

Btw, Akka Projection is a separate project and may not be in use at all. You can use Akka Persistence with this plugin without even using Akka Projection.